Editor/Window/GameLiftRequestAdapter.cs

```diff
@@ -0,0 +1,242 @@
+                    Locations = new List<LocationConfiguration>
+                    {
+                        new()
+                        {
+                            Location = fleetLocation
+                        }
+                    },
+                    Tags = new List<Tag>
+                    {
+                        new()
+                        {
+                            Key = "CreatedBy",
+                            Value = "AmazonGameLiftUnityPlugin"
+                        }
+                    }
+                    };
+                var createFleetResponse = await _gameLiftConfig.GameLiftWrapper.CreateFleet(createFleetRequest);
+                
+                if (createFleetResponse.HttpStatusCode == HttpStatusCode.OK)
+                {
+                    Debug.Log($"Created Fleet {fleetName}");
+                }
+                
+                return createFleetResponse.FleetAttributes.FleetId;
+            }
+            catch (Exception ex)
+            {
+                var errorBox = _container.Q<VisualElement>("FleetErrorInfoBox");
+                errorBox.style.display = DisplayStyle.Flex;
+                errorBox.Q<Label>().text = ex.Message;
+                Debug.Log(ex.Message);
+                return null;
+            }
+        }
+
+        internal async Task<List<FleetAttributes>> ListFleets()
+        {
+            try
+            {
+                var listFleetRequest = new ListFleetsRequest();
+                var listFleetResponse = await _gameLiftConfig.GameLiftWrapper.ListFleets(listFleetRequest);
+
+                var describeFleetRequest = new DescribeFleetAttributesRequest()
+                {
+                    FleetIds = listFleetResponse.FleetIds
+                };
+
+                var describeFleetResponse = await _gameLiftConfig.GameLiftWrapper.DescribeFleets(describeFleetRequest);
+                return describeFleetResponse.FleetAttributes;
+            }
+            catch (Exception ex)
+            {
+                var errorBox = _container.Q<VisualElement>("FleetErrorInfoBox");
+                errorBox.style.display = DisplayStyle.Flex;
+                errorBox.Q<Label>().text = ex.Message;
+                Debug.Log(ex.Message);
+                return null;
+            }
+        }
+
+        internal async Task<bool> RegisterFleetCompute(string computeName, string fleetId, string fleetLocation,
+            string ipAddress)
+        {
+            var webSocketUrl = await RegisterCompute(computeName, fleetId, fleetLocation, ipAddress);
+            if (webSocketUrl != null)
+            {
+                var computeNameResponse = _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.ComputeName, computeName);
+                var ipAddressResponse = _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.IpAddress, ipAddress);
+                var webSocketResponse = _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.WebSocketUrl, webSocketUrl);
```

Comment:
Could you help me understand what `PutSetting` is doing? 

I would expect the GameLiftRequestAdapter to just perform the API call logic, and the caller of a method in `GameLiftRequestAdapter` to use that fetched data to perform some operations, and not couple the both if possible.

Reply:
PutSetting saves the value to the local yaml file